ports/nrf/common-hal/analogio/AnalogIn.c

```diff
@@ -29,11 +29,21 @@
 #include "py/runtime.h"
 #include "supervisor/shared/translate.h"
 
-#include "nrfx_saadc.h"
+#include "nrf_saadc.h"
 #include "nrf_gpio.h"
 
 #define CHANNEL_NO 0
 
+void analogin_reset(void) {
+    // Calibrate the ADC once, on startup.
```

Comment:
_reset is called after each vm run. Do you want this in _init instead?

Reply:
That was unintentional, but the calibration can vary based on temperature, so maybe it's a good idea to do it on each vm run. It takes maybe a msec or so.

Reply:
Alternatively, we can call it once, and add a `calibrate()` function which is a no-op on other ports. I actually didn't notice any difference in the results before and after calibration, though.

Reply:
OK, I see some other calibration possibilities for other chips. I'll change it to `analogin_init()` and maybe add a `calibrate()` function later.